### PR TITLE
homework.lambda.part3.exercise2 (yushina)

### PR DESCRIPTION
**Pull request by AnyaYu**

### CODE REVIEW

**Review by vkovrovEPAM — APPROVED** [review]

**Comment by vkovrovEPAM** [comment]
Please resolve conflicts

**Comment by AnyaYu** [comment]
> Please resolve conflicts

Done

File: src/test/java/lambda/part3/exercise/Exercise2.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+         * Создает объект для маппинга, передавая ему новый список, построенный на основе исходного.
+         * Для добавления в новый список каждый элемент преобразовывается с использованием заданной функции.
+         * ([T], (T -> R)) -> [R]
+         * @param mapping Функция преобразования элементов.
+         */
+        public <R> MapHelper<R> map(Function<T, R> mapping) {
+            List<R> newList = new ArrayList<>();
+            source.forEach((s) ->
+                    newList.add(mapping.apply(s)));
+            return MapHelper.from(newList);
```
Comment (milkplusplus) [inline src/test/java/lambda/part3/exercise/Exercise2.java:46]: You can use `return from(newList);` here.

File: src/test/java/lambda/part3/exercise/Exercise2.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+         * Создает объект для маппинга, передавая ему новый список, построенный на основе исходного.
+         * Для добавления в новый список каждый элемент преобразовывается в список с использованием заданной функции.
+         * ([T], (T -> [R])) -> [R]
+         * @param flatMapping Функция преобразования элементов.
+         */
+        public <R> MapHelper<R> flatMap(Function<T, List<R>> flatMapping) {
+            List<R> newList = new ArrayList<>();
+            source.forEach((t) ->
+            newList.addAll(flatMapping.apply(t)));
+            return MapHelper.from(newList);
```
Comment (milkplusplus) [inline src/test/java/lambda/part3/exercise/Exercise2.java:59]: As well as here.

**Review by milkplusplus — APPROVED** [review]

**Pull request merged**